src/gtk/glcanvas.cpp

```diff
@@ -160,15 +167,16 @@ wxGLCanvas::wxGLCanvas(wxWindow *parent,
 
 static bool IsAvailable()
 {
+#if defined(GDK_WINDOWING_WAYLAND) && wxUSE_GLCANVAS_EGL
+    if ( GDK_IS_WAYLAND_DISPLAY(gdk_display_get_default()) )
+        return true;
+#endif
 #ifdef GDK_WINDOWING_X11
-    if ( !GDK_IS_X11_DISPLAY(gdk_display_get_default()) )
+    if ( GDK_IS_X11_DISPLAY(gdk_display_get_default()) )
+        return true;
 #endif
-    {
-        wxSafeShowMessage(_("Fatal Error"), _("wxGLCanvas is only supported on X11 currently.  You may be able to\nwork around this by setting environment variable GDK_BACKEND=x11 before starting\nyour program."));
-        return false;
-    }
-
-    return true;
+    wxSafeShowMessage(_("Fatal Error"), _("wxGLCanvas is only supported on Wayland (when using EGL backend) and X11 currently.  You may be able to\nwork around this by setting environment variable GDK_BACKEND=x11 before starting\nyour program."));
```

Comment:
I'm afraid this message is not as helpful as it could be for the end users. At the very least, we should somehow say that EGL must be available, e.g. "install EGL libraries and rebuild".

But for maximum clarity I think we need different messages for the two `#if` blocks above, i.e. when using Wayland we should just say "This program wasn't compiled with EGL support required under Wayland, either install EGL libraries and rebuild or run it under X11 backend by setting environment variable GDK_BACKEND=x11 ...".

Reply:
See what you think about the messages in my fixup.